File: backend/transaction-service/transaction-domain/transaction-domain-core/src/main/java/com/saver/system/transaction/service/domain/TransactionDomainService.java
package com.saver.system.transaction.service.domain;

import com.saver.system.transaction.service.domain.entity.Transaction;
import com.saver.system.transaction.service.domain.event.TransactionCreatedEvent;

public interface TransactionDomainService {
    TransactionCreatedEvent validateAndInitiateTransaction(Transaction transaction);
}

File: backend/Services/transactions/src/domain/domain-core/transaction-domain-service.ts
import { injectable } from "inversify";
import { Transaction } from "./entity/transaction";
import { TransactionCreatedEvent } from "./event/transaction-created-event";
import { ITransactionDomainService } from "./itransaction-domain-service";
@injectable()
export class TransactionDomainService implements ITransactionDomainService {
  public validateAndInitiateTransaction(
    transaction: Transaction
  ): TransactionCreatedEvent {
    transaction.validate();

    return new TransactionCreatedEvent(transaction);
  }
}
